Update to slicer 5

Connor-Bowley opened the pull request:
- Updates to use Slicer 5.
- Updates needed extensions to support that.
- Adds tutorials for new s-rep modules.
- Removes old s-rep modules.

Relates https://github.com/Kitware/SlicerSALT/issues/258

@jcfr I noticed in https://github.com/slicersalt/GROUPS there is a branch [spharmpdm-2021-05-43e7eef5f](https://github.com/slicersalt/GROUPS/tree/spharmpdm-2021-05-43e7eef5f) which has new changes. Do these need to be brought into SlicerSALT?

bpaniagua reviewed (approved):
lgtm

jcfr commented:
> Do these need to be brought into SlicerSALT?

Since all non SlicerSALT specific changes have been integrated upstream (we now have merge access), that specific branch is not relevant. See https://github.com/NIRALUser/GROUPS/commits/master

To move forward, you would need to create a new branch called `slicersalt-2022-04-29-9a72008c6` based of https://github.com/NIRALUser/GROUPS/commit/9a72008c6cbffa036e13c2dee18e3a2a23bc5d7c and cherry-pick these commits on top:


* [[slicersalt] ENH: Rename Rigid Alignment extension](https://github.com/slicersalt/GROUPS/commit/95fd339a0fe90dacc478a6c879728270b94ba84b)
* [[slicersalt] ENH: Disable building of non-used extensions](https://github.com/slicersalt/GROUPS/commit/d3cdb2427f47b32cc1c334e9ee437e1d3b24cb17)

Note that the following commit can *NOT* be cherry-picked as is:
* [[slicersalt] ENH: Update RigidAlignment](https://github.com/slicersalt/GROUPS/commit/bdd9129fee6dcff237347ce2f1a55d91e13bb537)

Indeed the same process should be done for [https://github.com/NIRALUser/RigidAlignment](RigidAlignment) so that the renaming specific to SlicerSALT is integrated on top of the latest version of `RigidAlignment`

jcfr commented:
### Module names

Since it looks like `SlicerCMF` and `SlicerSALT` expect the names of these modules to be different, we should introduce options allowing to configure these.

When the SlicerSALT change adds a category when there were no categorization (e.g https://github.com/slicersalt/RigidAlignment/commit/a12001cfa1b2f1ab6ddcdc0057dcb6178d97db58), I suggest to integrate the change upstream. 

### GROUPS options

Let's introduce option to avoid having to comment the code like it is currently done in https://github.com/slicersalt/GROUPS/commit/d3cdb2427f47b32cc1c334e9ee437e1d3b24cb17

jcfr reviewed (changes requested):
To ensure the project build on recent system, dependent extensions need to be updated.

Connor-Bowley commented:
> Let's introduce option to avoid having to comment the code like it is currently done in https://github.com/slicersalt/GROUPS/commit/d3cdb2427f47b32cc1c334e9ee437e1d3b24cb17

@vicory Would it be bad to just include these as well? Why are they excluded?

Connor-Bowley commented:
> Since it looks like SlicerCMF and SlicerSALT expect the names of these modules to be different, we should introduce options allowing to configure these.

Would it be worthwhile to have some kind of "module name override" mechanism in Slicer core that we could invoke from a custom app's home module?

Connor-Bowley commented:
@jcfr It seems you already added options for both the naming and the selective module build, so I will just use those.
https://github.com/NIRALUser/GROUPS/pull/30
https://github.com/NIRALUser/GROUPS/pull/29

Connor-Bowley commented:
Updated to use new GROUPS module. I needed to make a change to build correctly with the groups module metadata file. I have pushed the changes to the SlicerSALT branch and have put out a [pull request in NIRALUser](https://github.com/NIRALUser/GROUPS/pull/36) to bring the change in there as well.

@vicory Please double check that the modules are named and categorized appropriately after a clean build.

Connor-Bowley commented:
I have updated to use a new branch in [slicersalt/GROUPS](https://github.com/slicersalt/GROUPS) that is properly updated with the changes that went into [NIRALUser/GROUPS](https://github.com/NIRALUser/GROUPS) as part of https://github.com/NIRALUser/GROUPS/pull/36.

I squashed all the update package changes into 1 commit.

@jcfr Please re-review.
@vicory Please build and double check that all the renaming of GROUPS and RigidAlignment modules came through properly. Also, please double check the s-rep tutorial link for all the s-rep modules. I added a link so the SRepCreator module can download the sample data, although it does not auto-populate the UI. That is something for probably a different PR since I think I will need to make changes to the SRepCreator module to do it.

Thanks.

jcfr reviewed (approved):
Thanks @Connor-Bowley for diligently going through all the comments and following up with updates.

💯🙏🚀
